viz/pkg/healthcheck/healthcheck.go

```diff
@@ -192,9 +182,123 @@ func (hc *HealthChecker) vizCategory() *healthcheck.Category {
 			WithRetryDeadline(hc.RetryDeadline).
 			WithCheckRPC(func(ctx context.Context) (*healthcheckPb.SelfCheckResponse, error) {
 				return hc.vizAPIClient.SelfCheck(ctx, &healthcheckPb.SelfCheckRequest{})
-			}))
+			}),
+	}, true)
+}
+
+// VizDataPlaneCategory returns a healthcheck.Category containing checkers
+// to verify the data-plane metrics in prometheus and the tap injection
+func (hc *HealthChecker) VizDataPlaneCategory() healthcheck.Category {
+
+	return *healthcheck.NewCategory(LinkerdVizExtensionDataPlaneCheck, []healthcheck.Checker{
+		*healthcheck.NewChecker("data plane namespace exists").
+			WithHintAnchor("l5d-data-plane-exists").
+			Fatal().
+			WithCheck(func(ctx context.Context) error {
+				if hc.DataPlaneNamespace == "" {
+					// when checking proxies in all namespaces, this check is a no-op
+					return nil
+				}
+				return hc.CheckNamespace(ctx, hc.DataPlaneNamespace, true)
+			}),
+		*healthcheck.NewChecker("data plane proxy metrics are present in Prometheus").
+			WithHintAnchor("l5d-data-plane-prom").
+			Warning().
+			WithRetryDeadline(hc.RetryDeadline).
+			WithCheck(func(ctx context.Context) (err error) {
+				pods, err := hc.getDataPlanePodsFromVizAPI(ctx)
+				if err != nil {
+					return err
+				}
+
+				// TODO: Check if prometheus is present
```

Comment:
This seems like an important TODO? Maybe make a follow-up issue for this since this could possibly hang if Prometheus is disabled.

It's similar to my comment [here](https://github.com/linkerd/linkerd2/pull/5627#discussion_r566935296). If we can check the values config, we can end these checks early.

Reply:
This will be taken care of in a follow-up next week 👍